### ISSUE TITLE
[CRITICAL] [Groups] Frontend (Part 2): Update the Group Chat creation flow UI and Details page

### ISSUE THREAD

**Issue by marcaaron** [Expensify/App#34927]
Doc Section: https://docs.google.com/document/d/1NcklRh6NogUWhKL0J5udTE8bok3_9ZoZ57JTllP-QJg/edit#bookmark=id.pa6oyw3cxv7q

**Holding on:**

- [x] https://github.com/Expensify/Expensify/issues/340997
- [x] https://github.com/Expensify/Expensify/issues/352995
- [x] https://github.com/Expensify/Expensify/issues/340999
- [x] https://github.com/Expensify/Expensify/issues/341005
- [x] https://github.com/Expensify/Expensify/issues/341001
- [x] https://github.com/Expensify/Expensify/issues/341002
- [x] https://github.com/Expensify/Expensify/issues/341004
- [x] https://github.com/Expensify/Expensify/issues/341003
- [x] https://github.com/Expensify/App/issues/32317

[Group Chats Design Doc](https://docs.google.com/document/d/1NcklRh6NogUWhKL0J5udTE8bok3_9ZoZ57JTllP-QJg/edit)

**High level section with mockups:** https://docs.google.com/document/d/1NcklRh6NogUWhKL0J5udTE8bok3_9ZoZ57JTllP-QJg/edit#bookmark=id.9fdzsieljz63

At this point, the latest app version has already migrated off of Group DMs and is creating Group Chats under the hood. We now need to add the ability to:

- Give a [custom group name](https://docs.google.com/document/d/1NcklRh6NogUWhKL0J5udTE8bok3_9ZoZ57JTllP-QJg/edit#bookmark=id.k5q1rumirh3q) during creation
- Upload a [custom avatar](https://docs.google.com/document/d/1NcklRh6NogUWhKL0J5udTE8bok3_9ZoZ57JTllP-QJg/edit#bookmark=kix.wukdeb9gtfq7)
- Add the [update member role page and set the roles](https://docs.google.com/document/d/1NcklRh6NogUWhKL0J5udTE8bok3_9ZoZ57JTllP-QJg/edit#bookmark=id.1tyrvw8lwwpx)

We will also allow people to do these things + invite members via the "Details" page when tapping on a Group chat.

By the end of this ticket we should be able to:

**Creation**

- Give a custom group name
    - Tapping/clicking the group name on the confirmation will push to a new page where editing is possible. The group name is pre-selected to optimize for adding a custom, unique name instead of the default
![2024-01-17_11-29-07](https://github.com/Expensify/Expensify/assets/32969087/42abb3ae-3fe7-4a26-ab72-5072f9e5e1ae)

- Upload a custom avatar
    - This should work similar to how uploading an avatar works for workspaces ([example](https://github.com/Expensify/App/blob/278d04e0689d6f012354d89ec19c640a1cd9dc54/src/pages/workspace/WorkspaceSettingsPage.js#L114-L143)) 
![2024-01-17_11-36-50](https://github.com/Expensify/Expensify/assets/32969087/a4efc189-599c-4062-b49e-eaf5b6393f0b)

Both of these will need to be staged/stored locally so that they can be passed to the API when we tap "Start group"

**Details view (accessible from tapping on the Avatar in the header of the chat)**

![2024-01-17_11-35-05](https://github.com/Expensify/Expensify/assets/32969087/e33b61a6-86a9-4199-8885-1a7220604459)

- Avatar can be updated from here (by anyone)
- The "Leave" button will allow a user to leave the Group (HOLD on API changes)
- The "Pin" button will be the same action as right-click to pin
- "Call" will generate a Google Meet link
- Share page - works the same as existing chats
- Private notes - same as existing chats
- Settings - same as existing chats **however** we can also change the Name of the report here (anyone)

![2024-01-17_11-41-20](https://github.com/Expensify/Expensify/assets/32969087/4831ba32-b4e2-48c0-b42c-97d8f58725d9)

**Members page**

![2024-01-17_11-42-33](https://github.com/Expensify/Expensify/assets/32969087/b81cbd82-fefa-439b-847b-12fe657122c0)

The design for this closely follows the design for the [Simplified Collect design doc](https://docs.google.com/document/d/1Qv6V4nRqIbZfzQML_saCbrAEqQVIBsiQ3HE1UO7RCxQ/edit) so should be held on those changes and/or worked on in parallel and we will have a shared component for both flows.

- We can see who is an admin and not in the Members page (pill tag)
- For admins, we can select multiple or all members (for bulk actions)

![2024-01-17_12-02-30](https://github.com/Expensify/Expensify/assets/32969087/13d060f5-f6c5-42e6-ada2-ca08052ad146)

- Selecting members will replace the "Invite members" button with the `n selected` button with dropdown (shown above)
- Note: The highlighting styles when members are selected.
- Tapping the "3 selected" opens the popover menu with these options:
    - `Remove members`
    - `Make admin` (when at least one member is selected)
    - `Make member` (when at least one admin is selected)
- Confirmation modal will be shown for "Remove members"

![2024-01-17_12-06-16](https://github.com/Expensify/Expensify/assets/32969087/24f5e8ce-7c74-4301-a973-f2904878c90a)

- For members (non-admins), we will see this:

![2024-01-17_11-54-15](https://github.com/Expensify/Expensify/assets/32969087/8df908a4-ca03-486e-bc94-becb89c5ecf2)

Tapping a row (not the checkbox) will bring them to these screens respectively:

**Admins**

![2024-01-17_11-58-05](https://github.com/Expensify/Expensify/assets/32969087/e2b1996e-b95b-44ae-99e0-5dcb4946cb91)

- "Remove from group" option will launch a confirm modal
- Role selector will launch a new screen like this:
![2024-01-17_12-01-06](https://github.com/Expensify/Expensify/assets/32969087/e75018ee-ce87-4c6b-959c-203191ff615e)
- Profile - takes you to the profile

**Members**

Just go direct to the profile screen for the user

![2024-01-17_12-00-19](https://github.com/Expensify/Expensify/assets/32969087/7723fdfd-99b8-441a-9746-780fb0e965be)

**Invite member button** 

Opens to a new screen:

![2024-01-17_11-49-54](https://github.com/Expensify/Expensify/assets/32969087/c2cbc6a6-1360-44fe-833a-97b2246db8b2)

- Multiple members can be invited
- "Invite" button shows when at least one member is staged
- Redirect back to the "Members" page on success with new members visible.

![2024-01-17_11-52-31](https://github.com/Expensify/Expensify/assets/32969087/f8da1b8a-cba5-4229-854a-d448ac070fad)

**Testing membership changes**

When a member is added/removed or their admin status changes we will display the following "system messages" and must confirm these messages are present and showing up correctly prior to launching these features.

![2024-01-17_12-08-19](https://github.com/Expensify/Expensify/assets/32969087/cf485236-3011-45c2-a15b-f8137620d399)

- a system message will clarify when someone is invited, e.g. “invited %displayName%”, visible to all participants
- Removing a member from a `Group` chat triggers a system message for all `Group` chat participants, e.g. “removed %displayName%”
- When someone is made an admin, or demoted back to a member, we’ll use a “whisper” style message so that person knows that their role changed, e.g. “made %displayName% a member” or “made %displayName% an admin”

**Comment by marcaaron** [comment]
Bumping to Weekly but it's still on HOLD pending more Auth changes and the first App PR. This issue will introduce all the features we need to complete the Group Chats feature.

**Comment by marcaaron** [comment]
Still on HOLD as we need to get the first PR merged. I think @waterim will help us here as well. But let me know if that has changed!

**Comment by marcaaron** [comment]
Making this a Daily since I think the first PR is close enough that we can begin to work on the second.

**Comment by waterim** [comment]
Hello, Im Artem from Callstack

**Comment by marcaaron** [comment]
Removing this issue from the scope to HOLD on https://github.com/Expensify/Expensify/issues/341000 (I have doubts we will complete that particular migration by May 1st). 

However, the issue here really needs to happen ASAP.

@waterim Thank you for your continued hard work here! 🙇  

We ideally need to have something in review by next week. Let me know if it will be possible. I am hopeful because the APIs for these other features we are layering on top of the Members page should be pretty simple in comparison to adding the ability to create a Group Chats together with the Group DMs.

**Comment by marcaaron** [comment]
Last PR still not deployed - but removing the HOLD since there is no need to wait 🚀

**Comment by danielrvidal** [comment]
@waterim do you have an update on this one? This is high priority so it would be great to make some steady progress.

**Comment by waterim** [comment]
@danielrvidal yes, already had a conversation with @marcaaron 
Started today and working till the end of this week before OOO and after @marcaaron will take it not to loose any time here

**Comment by waterim** [comment]
My update: added invite, updated members page, added edit chat name

**Comment by marcaaron** [comment]
@s77rt would you like to work with us on this one as the C+?

**Comment by s77rt** [comment]
Yes sure!

**Comment by marcaaron** [comment]
The final major front end PR is more or less shaping up into a testable state 🚀 

**To do list before merge:**
- [ ] Wait for https://github.com/Expensify/Web-Expensify/pull/41571 to go to production to fix a bug related to updating the avatars (it's on staging already so should be sometime today or tomorrow).
- [ ] Trigger an AdHoc build on the current PR and loop in Design team for feedback.
- [ ] Cover failure cases for the new API commands. The "success" case works well, but there's basically no error handling so I want to make sure there is at least some minimal error handling in case something goes wrong.
- [ ] The es translations are placeholder. Need to request help with that.
- [ ] We need to confirm the marketing copy for this: 

    >In the case where the last member of a `Group` chat opts to leave, we’ll display a confirmation warning message explaining that the `Group` chat will be inaccessible to all users for all time if they leave.

Lastly, as we work through this it has become more clear that there are opportunities to harmonize the UI for both the UCR and Group Chat flows. I suspect we will have some follow up issues as part of this. My goal for this week is to get the Group Chats feature as laid out in the doc onto staging. Then we can look at everything else. cc @shawnborton @dannymcclain since you both had some ideas [here](https://expensify.slack.com/archives/C05RECHFBEW/p1712371681666529). I want to make sure we address them - but maybe we can do some follow up collaboration once we first implement what was decided in the Design Doc.

**Comment by marcaaron** [comment]
In final reviews now.

**Comment by marcaaron** [comment]
This is on production now.
